molgenis/molgenis-emx2
# fix: failed to create blanco database after selecting a different template (+ related fixes)

What are the main changes you did:

Using http://localhost:8080/apps/molgenis-components/#/component/InputSelect & http://localhost:8080/apps/central/#/ to explain changes.

### Old behavior
1) `required` showed the first element was automatically selected, yet no `modelvalue` was set and the first element could only be selected by first selecting another option and then selecting the first one again.

https://github.com/molgenis/molgenis-emx2/assets/7658246/f8f52384-8241-4631-8157-2d11a20107f6


2) If no `required` & `v-model=null`, `modelvalue` is not set. However, if selecting another option and the selecting the empty option again, it becomes `modelvalue=""` instead. 

https://github.com/molgenis/molgenis-emx2/assets/7658246/87aaf9bb-8c11-4f06-a043-9d8fcea3ac96

This caused issues in f.e. Database creation with no template:

https://github.com/molgenis/molgenis-emx2/assets/7658246/795c5491-3628-4a94-b7c2-783196cef2d3

![Scherm­afbeelding 2024-06-27 om 18 49 52](https://github.com/molgenis/molgenis-emx2/assets/7658246/dfcbb190-6cc5-4eb5-ba5f-6bc2e01e90ac)
![Scherm­afbeelding 2024-06-27 om 18 50 18](https://github.com/molgenis/molgenis-emx2/assets/7658246/9d902c12-ef33-4706-bb8b-4b0b5c4140ba)

3) There was a duplicate empty value when creating a new Database.
![Scherm­afbeelding 2024-06-27 om 19 04 04](https://github.com/molgenis/molgenis-emx2/assets/7658246/ba2c760f-cb71-4319-b16f-7131fd13de8e)


### New behavior
1) If `required`, an `$emit("update:modelValue"`  is triggered at `<select :modelValue=` to update it to the first item in `options` instead (simply returning the new value without an update showed the outdated value still in HTML).



2) If a `@change` is detected with a `$event.target.value ==''`, `$emit("update:modelValue", null)` is done instead of using the actual value. This causes the `modelvalue` attribute to be removed from `select`, similar to if no value was selected yet.

https://github.com/molgenis/molgenis-emx2/assets/7658246/8c83380d-ac16-4e58-bd4f-2af2a80fd946


3) Removed the `null` from `options` when creating a new Database, as a placeholder option is already generated when not using `required`.


how to test:
- Go to http://localhost:8080/apps/molgenis-components/#/component/InputSelect
- Validate whether the value shown after `Selected: ` is equal to what is actually selected in the drop-down menu.
- Validate whether the value shown after `Selected: ` is equal to the `modelvalue` as shown in the HTML (browser developer tools).
- Check using browser developer tools whether `modelvalue` attribute is not present if the blank value is selected, even after selecting other values.
- When creating a new database with blank template, it should still work even after having selected a different template.

todo:
